core/src/main/java/org/infinispan/util/CacheInjectable.java

```diff
@@ -0,0 +1,15 @@
+package org.infinispan.util;
+
+import org.infinispan.Cache;
+
+/**
+ * Interface that describes how a cache can be injected into another object.  This is useful for cases such as
+ * after an object is deserialized and you must inject a Cache into it.
+ */
+public interface CacheInjectable<K, V> {
```

Comment:
I wasn't planning on putting it into stream as it is not really specific to that.  I was thinking we could utilize that for other purposes as well.  But I guess if we don't think we ever will that is fine.